### DISCUSSION TITLE
Reconsider Gogs support in next version

### DISCUSSION THREAD

**Discussion by ogarcia** [woodpecker-ci/woodpecker#2161] [Request a Feature]
### Clear and concise description of the problem

As we can read in the documentation the next version will drop Gogs support. The truth is that it would be very interesting to reconsider this, let me explain.

It is true that nowadays Gogs is widely surpassed by Forgejo, but it is also true that if what you need is a simple web interface without too many frills, it is preferable to use Gogs rather than Forgejo. Gogs is much simpler and for those of us who have a local instance and don't need more than a few repositories, Gogs is the best choice. In such cases, using Forgejo would be "shooting flies with cannons".

### Suggested solution

Don't drop Gogs support in next version.

### Alternative

The truth is that there are not many other alternatives with the power of Gogs that are just as easy to use. It is true that there is cgit and gitolite, which together could be an alternative, but I think it is much easier to maintain an integration with Gogs (which in the end will be very similar to that of Forgejo) than one with gitolite.

### Additional context

This is obviously a suggestion for consideration. I have created this issue because I have not found a similar one and because, if you do not see this need, it would be answered.

### Validations

- [X] Checked that the feature isn't part of the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by qwerty287** [comment]
Gogs support was broken in 0.15 already, and we did not get any reports about issues there, so nobody seems to use it.
In addition to this, Gogs lacks support of *many* features which are more or less required for us, especially OAuth2. Adding Gogs support would require us to reintroduce a system to log in without OAuth2. Other features like commit statuses are not necessary, but still lacking.
Then there's the maintainer problem, because obviously nobody of us is using Gogs and thus nobody really will fix it if there are bugs or breaking changes. We have a similar problem with Bitbucket already, but there are some people that help fixing it from time to time.

> Gogs is […] for those of us who have a local instance and don't need more than a repository […] the best choice. In such cases, using Forgejo would be "shooting flies with cannons".

Maybe this is not correct or you have another view on this, but I think using Woodpecker on a single-repo local instance would be "shooting flies with cannons" too. Do you really need a full-featured CI system for this?

**Comment by 6543** [comment]
I dont mind adding gogs back, asap it does provide us Oauth2 and the basic apis ...

**Comment by ogarcia** [comment]
> Maybe this is not correct or you have another view on this, but I think using Woodpecker on a single-repo local instance would be "shooting flies with cannons" too. Do you really need a full-featured CI system for this?

It was a "lost in translation", I meant a few repositories. But it is also not about the number of repositories and the number of users. It's more about functionality. With Gogs you can have hundreds of repositories and users. To keep track of code and have a list of issues, you really don't need more (IMHO).

> I dont mind adding gogs back, asap it does provide us Oauth2 and the basic apis ...

Would it be enough for Gogs to provide oauth or would something else be necessary? I can open an issue in its repository and comment on it, no harm in trying.

**Comment by qwerty287** [comment]
Oauth2 is the most important part. Everything else should provide features which are not required to use, but some features will just not work at Woodpecker (especially commit statuses, pull requests, crons).
